Provisioning persistent volumes using OpenStack Manila

tsmetana opened the pull request:
This is the first version of the Manila external provisioner documentation.

@tmorriso-rh PTAL

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+====
+
+{product-title} is capable of provisioning persistent volume (PV) using
+link:https://wiki.openstack.org/wiki/Manila[OpenStack Manila] shared filesystem service.
```
s/filesystem/file system

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+[[start-controller-and-provisioner]]
+=== Starting the External Provisioner
+
+The external provisioner service is distributed as container image and can be run in the
```
s/distributed as container image/distributed as a container image

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+=== Starting the External Provisioner
+
+The external provisioner service is distributed as container image and can be run in the
+{product-title} cluster as usual. There is also a RPM versions for provisioner.
```
s/also a RPM versions/are also RPM versions

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+the provisioner will create for the volume. The `zones` parameter is a set of
+Manila availability zones the volume might be created in.
+
+The provisioner must be able to connect, authenticate and authorize to the
```
s/authenticate and/authenticate, and

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+
+[NOTE]
+====
+Newer OpenStack versions use "project" instead of "tenant" however the
```
s/"tenant" however/"tenant," however,

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+names.
+====
+
+The last step is to start the provisioner itself, e.g. using a deployment:
```
s/e.g./e.g.,

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+[[usage]]
+== Usage
+
+Once the provisioner is running the `PersistentVolumes` can be provisioned using
```
s/is running the/is running, the

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+  storageClassName: manila-share
+----
+
+The `PersistentVolumeClaim` is then bound to a `PersistentVolume` backed by
```
s/backed by/backed by the

s/is deleted/are deleted,

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,268 @@
+The `PersistentVolumeClaim` is then bound to a `PersistentVolume` backed by
+newly provisioned Manila share. When the `PersistentVolumeClaim` and
+subsequently the `PersistentVolume` is deleted the provisioner deletes and unexports
+also the Manila share.
```
s/also the Manila share/the Manila share

tmorriso-rh commented:
@tsmetana I suggested some edits in the persistent_storage_manila.adoc.

tsmetana commented:
@tmorriso-rh thanks for the review. I corrected the text and added a note mentioning only the NFS shares are supported: [bug #1573843](https://bugzilla.redhat.com/show_bug.cgi?id=1573843)

tmorriso-rh commented:
Thanks @tsmetana. Has this PR been reviewed by QE?

tsmetana commented:
[The Trello card](https://trello.com/c/2GAHIEmX/350-13-implement-dynamic-provisioner-for-manila-external-provisioner) has been moved to "accepted" by QE. I assume that includes the documentation review. @jhou1 can confirm eventually (thanks!)...

jhou1 commented:
I have verified the contents of the doc, I think it's well documented now.

tmorriso-rh commented:
@tsmetana Is this for 3.10 only?

tmorriso-rh commented:
@openshift/team-documentation PTAL.

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+
+[IMPORTANT]
+====
+Persistent volume Provisionining using OpenStack Manila is a Technology Preview
```
s/Provisioning/provisioning

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+endif::[]
+====
+
+{product-title} is capable of provisioning PV using the
```
s/PV/PVs

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+=== Starting the External Provisioner
+
+The external provisioner service is distributed as a container image and can be run in the
+{product-title} cluster as usual. There is also RPM versions for provisioner.
```
s/There is also RPM versions/There are also RPM versions available for the provisioner.

tsmetana replied:
The RPMs live in one of the RHEL channels. But since we would like the users to use only the containers in productions, I will not mention the RPMs.

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+The `type` parameter specifes the
+link:https://docs.openstack.org/manila/latest/admin/shared-file-systems-share-types.html[Manila share type]
+the provisioner will create for the volume. The `zones` parameter is a set of
+Manila availability zones the volume might be created in.
```
s\zones the\zones that the

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+
+The provisioner must be able to connect, authenticate, and authorize to the
+Manila service. The connection is configured using environment variables. There
+are several ways to do so. The following environment variable combinations might
```
s/There are several ways to do so./There are several ways to configure the connection.

Question: Is there a particular configuration that is recommended?

tsmetana replied:
No. It really depends on what the user knows/wants. It's the same as OpenStack clients.

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+----
+
+It is recommended to use `Secret` to pass the variables to the provisioner.
+The example of the `Secret` for the first variables combination:
```
s/The example of the `Secret` for the first variables combination:/. Example of the `Secret` for the first variables combination

tsmetana replied:
The `Secret` is used by an example below. So it's not only the `Secret`. Caption would be confusing I think.

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+
+The `PersistentVolumeClaim` is then bound to a `PersistentVolume` backed by
+the newly provisioned Manila share. When the `PersistentVolumeClaim` and
+subsequently the `PersistentVolume` are deleted the provisioner deletes and unexports
```
s/are deleted/are deleted,

tmorriso-rh commented:
@kalexand-rh Thanks for the excellent peer review. If there are no additional suggestions, can you add the Peer Review Done label? Thanks.

tsmetana commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+{product-title} is capable of provisioning PV using the
+link:https://wiki.openstack.org/wiki/Manila[OpenStack Manila] shared file system service.
+
+It is assumed the OpenStack Manila service has been correctly set up and is
```
The style has been copied from other documents (English is not my mother tongue, so I steal a lot when writing documentation) and in this case I would say the passive form is actually appropriate: the Manila service simply has to be set up by someone somewhere and it's not our user who should do it or even care about it.

tsmetana commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+To allow the containers managing the API objects, the necessary role-based access control
+(RBAC) rules need to be configured by the administrator:
+
+. Create a `ServiceAccount`:
```
Added link to RBAC authorization concept documentation to the overview.

tsmetana commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+
+The `type` parameter specifes the
+link:https://docs.openstack.org/manila/latest/admin/shared-file-systems-share-types.html[Manila share type]
+the provisioner will create for the volume. The `zones` parameter is a set of
```
The 'list' is a data type and its representation is defined in the YAML syntax. Since there are two ways to express it I would rather leave it up to the user to choose.

tsmetana commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+are several ways to do so. The following environment variable combinations might
+be used:
+
+[source]
```
These are only the variable names and the usage is shown in the `secret` example: i.e. they have to be set but (as described) in the provisioner's environment -- in the container.

tsmetana commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+OS_TENANT_ID
+----
+
+It is recommended to use `Secret` to pass the variables to the provisioner.
```
True... I re-worded the sentence.

tsmetana commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+It is recommended to use `Secret` to pass the variables to the provisioner.
+The example of the `Secret` for the first variables combination:
+
+[source,yaml]
```
`oc create -f <the yaml file>` -- it's not different from any other API object described in the documentation so I don't think it needs extra description.

tsmetana commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,270 @@
+
+The last step is to start the provisioner itself, e.g. using a deployment:
+
+[source, yaml]
```
These are all OpenShift API objects that need to be created using `oc create`. I don't see this being described in the other documents and the overview section suggests to get familiar with the basic concepts anyway.

tsmetana commented:
Thanks for the reviews. I have pushed an updated version. PTAL.

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,265 @@
+<2> Set of Manila availability zones that the volume might be created in.
+
+Configure the provisioner to connect, authenticate, and authorize to the Manila service by using
+environment variables. Select the right combination of environment variables for your installation
```
s/right/appropriate

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,265 @@
+----
+
+To pass the variables to the provisioner, use a `Secret`.
+Example of the `Secret` for the first variables combination:
```
Suggest changing this to "The following example shows a `Secret` configured for the first variables combination":

tmorriso-rh commented on install_config/persistent_storage/persistent_storage_manila.adoc:
```diff
@@ -0,0 +1,265 @@
+[[usage]]
+== Usage
+
+After the provisioner is running you can provision PersistentVolumes by using a
```
Suggest changing this to "After the provisioner is running, you can provision PVs using a PVC and the corresponding StorageClass:"

tmorriso-rh commented:
@tsmetana I suggested a few additional changes. PTAL. Thanks.

tmorriso-rh commented:
/cherrypick enterprise-3.10
